source/DSCResources/DSC_CMDistributionPoint/DSC_CMDistributionPoint.psm1

```diff
@@ -0,0 +1,648 @@
+        MinimumFreeSpaceMB              = $freespace
+        PrimaryContentLibraryLocation   = $availcontentDrivePrimary
+        SecondaryContentLibraryLocation = $availContentDriveSecondary
+        PrimaryPackageShareLocation     = $availPkgSharePrimary
+        SecondaryPackageShareLocation   = $availPkgShareSecondary
+        ClientCommunicationType         = $clientCommType
+        BoundaryGroups                  = $bGroups
+        AllowPreStaging                 = $dpInfo.PreStagingAllowed
+        CertificateExpirationTimeUtc    = $validData
+        EnableAnonymous                 = $anonymous
+        EnableBranchCache               = $branchCache
+        EnableLedbat                    = $dpInfo.EnableLEDBAT
+        Ensure                          = $status
+    }
+}
+
+<#
+    .SYNOPSIS
+        This will set the desired state.
+
+    .PARAMETER SiteCode
+        Specifies the site code for Configuration Manager site.
+
+    .PARAMETER SiteServerName
+        Specifies the SiteServer to install the role on.
+
+    .PARAMETER Description
+        Specifies a description for the distribution point.
+
+    .PARAMETER MinimumFreeSpaceMB
+        Specifies the amount of free space to reserve on each drive used by this distribution point.
+        Only used when distribution point is not currently installed.
+
+    .PARAMETER PrimaryContentLibraryLocation
+        Specifies the primary content location. Configuration Manager copies content to the primary content location
+        until the amount of free space reaches the value that you specified.
+        Only used when distribution point is not currently installed.
+
+    .PARAMETER SecondaryContentLibraryLocation
+        Specifies the secondary content location.
+        Only used when distribution point is not currently installed.
+
+    .PARAMETER PrimaryPackageShareLocation
+        Specifies the primary package share location. Configuration Manager copies content to the primary package
+        share location until the amount of free space reaches the value that you specified.
+        Only used when distribution point is not currently installed.
+
+    .PARAMETER SecondaryPackageShareLocation
+        Specifies the secondary package share location.
+        Only used when distribution point is not currently installed.
+
+    .PARAMETER CertificateExpirationTimeUtc
+        Specifies, in UTC format, the date and time when the certificate expires.
+        If not specified and a Distribution Point is added, by a certificate will be
+        generated with an expiration date of 2 years from date installed.
+
+    .PARAMETER ClientCommunicationType
+        Specifies protocol clients or devices communicate with the distribution point.
+
+    .PARAMETER BoundaryGroups
+        Specifies an array of boundary groups by name.
+
+    .PARAMETER BoundaryGroupStatus
+        Specifies if the boundary group is to be added, removed, or match BoundaryGroups.
```

Comment:
Specify add/remove/match on the DP, so no one gets confused and thinks this will create boundary groups.

Reply:
I misunderstood the intent. This works.